dts/bindings/display/solomon,ssd16xx-common.yaml

```diff
@@ -0,0 +1,105 @@
+# Copyright (c) 2018, Phytec Messtechnik GmbH
+# SPDX-License-Identifier: Apache-2.0
+
+description: SSD16XX EPD display controller
+
+include: [spi-device.yaml, display-controller.yaml]
+
+properties:
+    softstart:
+      type: uint8-array
+      required: false
+      description: Booster soft start values
+
+    orientation-flipped:
+      type: boolean
+      required: false
+      description: Last column address is mapped to first segment
+
+    reset-gpios:
+      type: phandle-array
+      required: true
+      description: RESET pin.
+
+        The RESET pin of SSD16XX is active low.
+        If connected directly the MCU pin should be configured
+        as active low.
+
+    dc-gpios:
+      type: phandle-array
+      required: true
+      description: DC pin.
+
+        The DC pin of SSD16XX is active low (transmission command byte).
+        If connected directly the MCU pin should be configured
+        as active low.
+
+    busy-gpios:
+      type: phandle-array
+      required: true
+      description: BUSY pin.
+
+        The BUSY pin of SSD16XX is active high.
+        If connected directly the MCU pin should be configured
+        as active high.
+
+    tssv:
+      type: int
+      required: false
+      description: Temperature Sensor Selection Value
+
+        Display controller can have integrated temperature sensor or
+        an external temperature sensor is connected to the controller.
+        The value selects which sensor should be used.
+
+child-binding:
+    description: |
+      Child nodes describe refresh profiles. Each refresh profile
+      specifies a set of voltages used to drive the refresh
+      cycle. Refresh profiles are optional and are used to override
+      defaults loaded from the controllers OTP memory.
+
+      Partial refresh will be disabled unless a partial refresh
+      profile has been specified. That profile may be empty to use the
+      defaults loaded from OTP if supported by the device.
+
+      The driver typically looks for the following child nodes:
+
+        - 'full' - Normal / full refresh.
+        - 'partial' - Partial refresh.
+
+    properties:
```

Comment:
This seems like a zephyr specific config, so I'd ask to add a `compatible` to the child node.  Something like `zephyr,ssd16xx-display-profile` and maybe look to utilize an additional compatible like `zephyr,-ssd16xx-display-profile-full` and `zephyr,-ssd16xx-display-profile-partial` (we probably don't have the ability to describe that properly in the binding beyond in the description field at this time).

Reply:
Is a compatible really necessary? There are a few cases where Linux uses device tree objects without a compatible. For example, display timings (https://elixir.bootlin.com/linux/latest/source/Documentation/devicetree/bindings/display/panel/display-timings.yaml) nodes don't use a compatible. I realise that Zephyr uses device trees in a subtly different way though, so I don't know what's works best with Zephyr's design philosophy.

I think I would like to avoid mode-specific compatibles since that won't scale. I can see reasons for adding support for more refresh profiles in the future. One possibility that I explored was to have a device-specific API to select a refresh profile. When trying that, I briefly considered (ab)using a reg property and providing well-known names for standard profiles such as full and partial. This seems a bit smelly though.

Another, potentially cleaner approach, would be to generate structs for all of the provided profiles and have a macro to refer to those instances using their node label. In this case, we could have a properties that point to the relevant default profiles (full/partial).

Reply:
Linux devicetree's have a regularity problem in that the same issue is solved different ways.  Since you are giving meaning to the profile, I think it makes sense to describe that with a `compatible` rather than saying the node name should be X or Y.  I don't know enough about the domain area of display choices to really say much more about how many different profiles you might have, and how you might choose between them.

Reply:
I think part of the problem here is that we potentially need more profiles than the current two. I can see the logic in having a single "zephyr,ssd16xx-refresh-profile" compatible if that makes works better for the tooling though. It's also not necessarily a Zephyr-specific config. The refresh profile describes a property of the hardware that the driver needs to know to configure the display correctly.

Would it be better to have named properties in the main device node that  points to specific well-known profiles (full/partial)? We could use a generic compatible (e.g., "solomon,ssd16xx-refresh-profile") for the profiles in that case.

Reply:
Since this PR is significant improvement, @galak @andysan  can we come to a solution for the next release?

Reply:
It's worth noting that https://github.com/zephyrproject-rtos/zephyr/pull/48383 has now been merged. If we end up changing the way profiles are handled here, we'll need to update that driver as well for consistency.

Reply:
I don't really care about the approach we use here whether it is a child node naming convention, a property that points to a profile, or something else. However, I do feel strongly about type-specific compatibles though. I don't believe that will work since I can see a path towards multiple profiles that can be selected at runtime (fast, fast differential, full, black, red, etc.). I don't have any patches planned to implement this in the near term though.

One option I was considering briefly was to add a 'reg' property to the sub node and provide well-known full/partial IDs in a header file. This would neatly scale to an arbitrary number of user-defined profiles.